tests/test_astrometryTask.py

```diff
@@ -99,18 +98,30 @@ def testUsedFlag(self):
                 count += 1
         self.assertEqual(count, len(results.matches))
 
-    def testMaxMeanDistance(self):
-        """If the astrometric fit does not satisfy the maxMeanDistanceArcsec
-        threshold, ensure task raises an lsst.pipe.base.TaskError.
+    def testWcsFailure(self):
+        """In the case of a failed WCS fit, test that the exposure's WCS is set
+           to None and the coord_ra & coord_dec columns are set to nan in the
+           source catalog.
         """
         self.exposure.setWcs(self.tanWcs)
         config = AstrometryTask.ConfigClass()
-        config.maxMeanDistanceArcsec = 0.0  # To ensure a "deemed" WCS failure
-        solver = AstrometryTask(config=config, refObjLoader=self.refObjLoader)
-        sourceCat = self.makeSourceCat(self.tanWcs, doScatterCentroids=True)
-
-        with self.assertRaisesRegex(pipeBase.TaskError, "Fatal astrometry failure detected"):
-            solver.run(sourceCat=sourceCat, exposure=self.exposure)
+        config.wcsFitter.order = 2
+        config.wcsFitter.maxScatterArcsec = 0.0  # To ensure a WCS failure
+        sourceSchema = afwTable.SourceTable.makeMinimalSchema()
+        measBase.SingleFrameMeasurementTask(schema=sourceSchema)  # expand the schema
+        # schema must be passed to the solver task constructor
+        solver = AstrometryTask(config=config, refObjLoader=self.refObjLoader, schema=sourceSchema)
+        sourceCat = self.makeSourceCat(self.tanWcs, sourceSchema=sourceSchema, doScatterCentroids=True)
+        results = solver.run(
+            sourceCat=sourceCat,
+            exposure=self.exposure,
+        )
```

Comment:
Is it also possible to test if we log a warning as expected?  (I actually don't know how to do that...)

Reply:
Yes. `with self.assertLogs()` (if it's a C++ lsst.log log message you have to use an additional bit of code to redirect the C++ logging to python logging)